src/Nethermind/Nethermind.Trie/PatriciaTree.cs

```diff
@@ -1254,21 +1283,38 @@ Hash256 DecodeStorageRoot(Hash256 root, Hash256 address)
                 rootHash = storageRoot ?? DecodeStorageRoot(rootHash, storageAddr);
             }
 
-            ReadFlags flags = visitor.IsFullDbScan
-                ? visitor.ExtraReadFlag | ReadFlags.HintCacheMiss
-                : visitor.ExtraReadFlag;
+            ReadFlags flags = visitor.ExtraReadFlag;
+            if (visitor.IsFullDbScan)
+            {
+                if (TrieStore.Scheme == INodeStorage.KeyScheme.HalfPath)
```

Comment:
I'd move this code elsewhere. It's dependent on the `TrieStore.`Scheme` so maybe TrieStore could provide `FullDbScanFlags`?

Reply:
Fair point. NodeStorage already change the readahead flag already.